sydneywait/Kanawaha-Trace
# Routes

I added the functionality on the "My Routes" page to edit a route once it is created in the "Explore" page.

On the "My Routes" page you will see two columns--planned routes and completed routes.  This will only show the planned and completed routes for the active user.  Each column has cards that show a trail marker with white/yellow/white or yellow/white/yellow to indicate the direction of travel on the trail.  This corresponds to the trail marker you will see in real life.  There is also a placeholder image of a map and elevation profile, as well as some text.

### Planned Routes.
This section shows cards with placeholder text and the following 4 buttons:
* reverse:  The name of the route will change as well as the image of the trail marker indicating the trail direction.  This logic sends a patch to the database to change the start and end points.

* complete: this button will engage a modal that allows you to input the date and time completed.
 
* delete.  You will be given a modal to ask if you are sure before confirming the delete.

 * details: this will take you to a details page that will eventually have information about the route hazards and features.  From this page you can complete (enter date and time) edit (change the start and end points) or delete the route.

### Completed Routes:
This section shows the completed date and time on the card, and the cards have only two buttons--details and delete.
* details will take you to a detail view of the route which lists the date and time completed, and will eventually list all the hazards and features of the route.  From this view you can edit the date and time completed or you can delete the route.
* the delete button will give you a modal asking for confirmation to delete before deleting the card.


### to test:
Use the 'explore' page to create multiple routes.  You can then experiment with deleting the routes, editing the route, completing the route, or looking at details.  Be sure to check that the buttons are rendering appropriately (completed routes should not have a 'complete' button, for example).  Double check the database after making changes to ensure that the patch, edit, post, and delete operations are working as expected.
